data_structures/sliding_window.cpp

```diff
@@ -0,0 +1,159 @@
+
+/*
+ * @brief Sliding window implementation
+ *
+ * @details A sliding window is a fixed-size FIFO data structure.
+ *         This implementation uses the vector class of the standard library.
```

Comment:
Maybe provide more information on how the algorithm works, what does it do, etc.. You can take it from Wikipedia and adjust it to our needs. Otherwise, if you just copy it like that, it would be plagiarism. :)